cloud/amazon/ec2_vol.py

```diff
@@ -326,9 +344,31 @@ def attach_volume(module, ec2, volume, instance):
             changed = True
         except boto.exception.BotoServerError, e:
             module.fail_json(msg = "%s: %s" % (e.error_code, e.error_message))
+        
+        # If 'delete on termination' is set then modify instance attribute for volume
+        if delete_on_termination:
+            modify_dot_attribute(module, ec2, instance, device_name)
```

Comment:
Maybe a little more logic is required here?  What if the volume is already attached but the flag is currently set to the opposite of what is specified?  You should also ensure that change state is consistent.  What happens if the volume exists, was attached but the flag was changed? We need to report back as changed=True